### PR TITLE
feat: Weekly calendar event border color

### PR DESCRIPTION
**Pull request by sean-brydon**
Smaller PR seperated from the stack that is safe to merge by its self. 

Adds basic support for custom colors for the up coming troubleshooter that uses calendar view

Safe PR that removes the times displayed for 30 mins events 

adds support for a custom border color in events 

Figma: https://www.figma.com/file/UajFu4M1APhn2AywAEJkJr/New-Features?type=design&node-id=5966-558456&mode=design

### CODE REVIEW

**Comment by sean-brydon** [comment]
Current dependencies on/for this PR:
* main
  * **PR #11907** <a href="https://app.graphite.dev/github/pr/calcom/cal.com/11907" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/calcom/cal.com/11907?utm_source=stack-comment).

File: packages/features/calendars/weeklyview/components/event/Event.tsx [hunk 1]
```diff
@@ -11,6 +12,35 @@ type EventProps = {
   disabled?: boolean;
 };
 
+const eventClasses = cva(
```
Comment (sean-brydon) [inline packages/features/calendars/weeklyview/components/event/Event.tsx:15]: Using CVA for this as its less complicated than what we had

File: packages/features/calendars/weeklyview/components/event/Event.tsx [hunk 2]
```diff
@@ -19,27 +49,32 @@ export function Event({
...
+        disabled,
+        selected,
+        borderColor,
+      })}
+      style={styles}>
       <div className="w-full overflow-hidden overflow-ellipsis whitespace-nowrap text-left leading-4">
         {event.title}
       </div>
-      {eventDuration >= 30 && (
+      {eventDuration > 30 && (
```
Comment (sean-brydon) [inline packages/features/calendars/weeklyview/components/event/Event.tsx:77]: 30 min time slots were being cut off from overflow. Removing the displayed time if the event is 30mins and only show > than

File: packages/features/calendars/weeklyview/components/event/Event.tsx [hunk 3]
```diff
@@ -19,27 +49,32 @@ export function Event({
...
   const selected = currentlySelectedEventId === event.id;
+  const { options } = event;
+
+  const borderColor = options?.borderColor ? "custom" : options?.status;
+
+  const styles = options?.borderColor
+    ? {
+        borderColor: options?.borderColor,
+      }
+    : {};
```
Comment (sean-brydon) [inline packages/features/calendars/weeklyview/components/event/Event.tsx:60]: Use a "custom" border prop that applies 0 styling L:38 

Allowing us to use the dynamic styles

**Review by Udit-takkar — APPROVED** [review]
LGTM. Great Work

**Pull request merged**